File: Atcoder_#360/ruby/b_vertical_reading.rb
input = gets.chomp.split
S = input[0]
T = input[1]

n = S.length
t_len = T.length

found = false

(1...n).each do |w|
  (1..w).each do |c|
    new_string = ""
    idx = c - 1

    while idx < n
      new_string << S[idx] if idx < n
      idx += w
    end

    if new_string == T
      found = true
      break
    end
  end

  break if found
end

puts found ? "Yes" : "No"

File: Atcoder_#360/javascript/b_vertical_reading.js
function main(input) {
  input = input.trim().split(' ');
  const S = input[0];
  const T = input[1];

  const n = S.length;
  const t_len = T.length;

  let found = false;

  for (let w = 1; w < n; w++) {
    for (let c = 1; c <= w; c++) {
      let new_string = "";
      let idx = c - 1;

      while (idx < n) {
        new_string += S[idx];
        idx += w;
      }

      if (new_string === T) {
        found = true;
        break;
      }
    }
    if (found) {
      break;
    }
  }

  console.log(found ? "Yes" : "No");
}

const fs = require('fs');
const input = fs.readFileSync('/dev/stdin', 'utf8');
main(input);
